src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
```

Comment:
The implementation of `apply` method in `PurchaseOperation` class should check if the result balance is positive before saving it to the Storage. If the balance is negative, it should throw a `RuntimeException` to prevent incorrect data from being saved.